wrappers/kn/src/nativeMain/kotlin/zxingcpp/BarcodeReader.kt

```diff
@@ -0,0 +1,113 @@
+	toKString().also { ZXing_free(this) }.ifEmpty { null }
+}
+
+
+@OptIn(ExperimentalForeignApi::class)
+class BarcodeReader : ReaderOptions() {
+	@Throws(BarcodeReadingException::class)
+	fun read(imageView: ImageView): List<Barcode> =
+		ZXing_ReadBarcodes(imageView.cValue, cValue)?.let { cValues -> cValues.toKObject().also { ZXing_Barcodes_delete(cValues) } }
+			?: throw BarcodeReadingException(ZXing_LastErrorMsg()?.toKStringNullPtrHandledAndFree())
+}
+
+class BarcodeReadingException(message: String?) : Exception("Failed to read barcodes: $message")
+
+
+@OptIn(ExperimentalForeignApi::class)
+open class ReaderOptions {
+	var tryHarder: Boolean
+		get() = ZXing_ReaderOptions_getTryHarder(cValue)
+		set(value) = ZXing_ReaderOptions_setTryHarder(cValue, value)
+	var tryRotate: Boolean
+		get() = ZXing_ReaderOptions_getTryRotate(cValue)
+		set(value) = ZXing_ReaderOptions_setTryRotate(cValue, value)
+	var tryDownscale: Boolean
+		get() = ZXing_ReaderOptions_getTryDownscale(cValue)
+		set(value) = ZXing_ReaderOptions_setTryDownscale(cValue, value)
+	var tryInvert: Boolean
+		get() = ZXing_ReaderOptions_getTryInvert(cValue)
+		set(value) = ZXing_ReaderOptions_setTryInvert(cValue, value)
+	var isPure: Boolean
+		get() = ZXing_ReaderOptions_getIsPure(cValue)
+		set(value) = ZXing_ReaderOptions_setIsPure(cValue, value)
+	var returnErrors: Boolean
+		get() = ZXing_ReaderOptions_getReturnErrors(cValue)
+		set(value) = ZXing_ReaderOptions_setReturnErrors(cValue, value)
+	var binarizer: Binarizer
+		get() = Binarizer.fromCValue(ZXing_ReaderOptions_getBinarizer(cValue))
+		set(value) = ZXing_ReaderOptions_setBinarizer(cValue, value.cValue)
+	var formats: Set<BarcodeFormat>
+		get() = ZXing_ReaderOptions_getFormats(cValue).parseIntoBarcodeFormat()
+		set(value) = ZXing_ReaderOptions_setFormats(cValue, value.toValue())
+	var eanAddOnSymbol: EanAddOnSymbol
+		get() = EanAddOnSymbol.fromCValue(ZXing_ReaderOptions_getEanAddOnSymbol(cValue))
+		set(value) = ZXing_ReaderOptions_setEanAddOnSymbol(cValue, value.cValue)
+	var textMode: TextMode
+		get() = TextMode.fromCValue(ZXing_ReaderOptions_getTextMode(cValue))
+		set(value) = ZXing_ReaderOptions_setTextMode(cValue, value.cValue)
+	var minLineCount: Int
+		get() = ZXing_ReaderOptions_getMinLineCount(cValue)
+		set(value) = ZXing_ReaderOptions_setMinLineCount(cValue, value)
+	var maxNumberOfSymbols: Int
+		get() = ZXing_ReaderOptions_getMaxNumberOfSymbols(cValue)
+		set(value) = ZXing_ReaderOptions_setMaxNumberOfSymbols(cValue, value)
+
+	val cValue: CValuesRef<ZXing_ReaderOptions>? = ZXing_ReaderOptions_new()
+
+	@Suppress("unused")
+	@OptIn(ExperimentalNativeApi::class)
+	private val cleaner = createCleaner(cValue) { ZXing_ReaderOptions_delete(it) }
+}
+
+@OptIn(ExperimentalForeignApi::class)
+enum class Binarizer(internal val cValue: ZXing_Binarizer) {
+	LocalAverage(ZXing_Binarizer.ZXing_Binarizer_LocalAverage),
```

Comment:
In the `ImageFormat` enum class, you did not require to fully qualify each value, so it should not be required in those enums here either, right? `ZXing_Binarizer.ZXing_Binarizer_LocalAverage` is just very verbose.

Reply:
Oh, I didn't notice this, this is modified by ide when I was moving these sources from another file

Reply:
fixed in https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/c56af6d8be841a58ce60ff6fc15563c0885409a9